Update usage string for VBC and update checklist for new LangVersion

jcouv opened the pull request:
**Customer scenario**
Type `vbc -?`
The usage string doesn't list language version 15.3 (but should).

Also addresses some suggestions from previous PR (making fields read-only, adjusting name of parameter). (FYI @cston)

**Risk**
**Performance impact**
Low. Just changing a resource string.

**Root cause analysis:**
I'd missed this when adding the new language version. This PR also updates the checklist for new language version, to avoid missing this in the future.

**How was the bug found?**
Ad-hoc testing.

@dotnet/roslyn-compiler for review.

cston commented on src/Compilers/VisualBasic/Portable/VBResources.resx:
```diff
@@ -5130,7 +5130,7 @@
 /langversion:&lt;number&gt;             Specify language version: 
                                   9|9.0|10|10.0|11|11.0|12|12.0|14|14.0|15|
-                                  15.0|default|latest
+                                  15.0|15.3|default|latest
```
Consider adding a test that `vbc -?` contains `LanguageVersion.GetErrorName()` for each value.

jcouv replied:
I added a test for `ToDisplayString`.
`GetErrorName` always includes a decimal, but `ToDisplayString` uses integral numbers for major versions.
Not sure which is best.

jaredpar commented:
retest this please

cston commented:
LGTM

jcouv commented:
@dotnet/roslyn-compiler for a second review. Tiny change. Thanks

jcouv commented:
@MeiChin-Tsai @jaredpar for ask-mode approval.

**Customer scenario**
Type `vbc -?`
The usage string doesn't list language version 15.3 (but should).

Also addresses some suggestions from previous PR (making fields read-only, adjusting name of parameter).

**Risk**
**Performance impact**
Low. Just changing a resource string.

**Root cause analysis:**
I'd missed this when adding the new language version. This PR also updates the checklist and test for new language version, to avoid missing this in the future.

**How was the bug found?**
Ad-hoc testing.

AlekseyTs commented on src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs:
```diff
@@ -1393,6 +1393,27 @@ public void LanguageVersion_TryParseDisplayString(string input, bool success, La
+
+            foreach (var version in expected)
+            {
+                Assert.True(helpRange.Contains(version), $"Missing version '{version}'");
```
>Assert.True(helpRange.Contains(version), $"Missing version '{version}'"); [](start = 16, length = 73)

It doesn't look like the check is strong enough. For example, it looks like it will find "7" in "7.1" and will not fail when "7" is not there.

jcouv replied:
Yes. Also, "1" matches "iso-1".
If we add "7.1" before "7", yes, we'll run into a problem. Also, if we add "7.8", then forget to add "8", we'll also run into a problem.

This check is not meant to remove the checklist, but to provide a reminder in common cases. I don't think it's worth it to fully parse the usage string. Let me know if you feel otherwise.

AlekseyTs commented on src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs:
```diff
@@ -1403,10 +1403,12 @@ public void LanguageVersion_CommandLineUsage()
             foreach (var version in expected)
             {
-                Assert.True(helpRange.Contains(version), $"Missing version '{version}'");
+                var foundIndex = helpRange.IndexOf(version);
```
It feels like we should make sure the previous character is a white-space, or a comma.

MeiChin-Tsai commented:
approved.
